osu.Game/Screens/Edit/Compose/Components/SelectionBox.cs

```diff
@@ -91,6 +91,23 @@ public bool CanScaleY
             }
         }
 
+        private bool canScaleProportionally;
+
+        /// <summary>
+        /// Whether vertical scaling support should be enabled.
+        /// </summary>
+        public bool CanScaleProportionally
```

Comment:
I know that this name comes from me but I just realised it kinda doesn't work for beatmap editor, because there you can resize non-proportionally...

Maybe `CanScaleDiagonally` instead...?

xmldoc needs updating btw. And may be worth adding a clarification why this is separated out.